File: python/simple-cipher/simple_cipher.py
import random
class Cipher:
    def __init__(self, key=None):
        if key == None:
            self.key = ''
            for i in range(0,100):
                self.key += chr(random.randint(97,122))
        else:
            self.key = key

        self.keyn=[]
        for ch in self.key:
                self.keyn.append(ord(ch)-ord('a'))

        self.kl = len(self.keyn)
        print (self.keyn)
        print (self.kl)

    def encode(self, text):
        encode_txt=''
        i = 0
        for ch in text:
            if ord(ch) + self.keyn[i % self.kl] <= 122:
                encode_txt += chr(ord(ch) + self.keyn[i % self.kl])
            else:
                encode_txt += chr(ord(ch) + self.keyn[i % self.kl] - ord('z')+ ord('a') - 1)
            i += 1
        return encode_txt

    def decode(self, text):
        decode_txt = ''
        i = 0
        for ch in text:
            print(ch,ord(ch),chr(97))
            if ord(ch) - self.keyn[i % self.kl] < 97:
                decode_txt += 'z'
            else:
                decode_txt += chr(ord(ch) - self.keyn[i % self.kl] - 1)
            i += 1
        return decode_txt

#cipher = Cipher("abcdefghij")
#print(cipher.decode("zabcdefghi"))
print('pppo',ord('a'), ord('z'), ord('r'),chr(106))

File: javascript/simple-cipher/simple-cipher.js
import { createHook } from "async_hooks";

export class Cipher {
    constructor(key) {
        this.alpha = 'abcdefghijklmnopqrstvuwxyz';
        if (key == undefined) {
            for (var i = 0; i<100; i++){
                this.key += chr(ord('a')+Math.random() * 26);
            };
            console.log('random key=',this.key)
        } else {
            this.key = key;
        }
    }

    encode (sentence) {
        var code =""
        for (var i = 0; i<sentence.length; i++) {
            var ch = sentence.substr(i,1);
            var codepos = ord(ch) - ord('a');
            if (codepos > this.key.length) codepos = codepos % this.key.length;
            code += key.substr(codepos,1);
        }
        return code;
    }

    decode (code) {
        var sentence = "";
        for (var i = 0; i<code.length; i++) {
            var ch = code.substr(i,1);
            var letpos = ord(ch) - ord('a');
            if (letpos > this.key.length) letpos = letpos % this.key.length;
            sentence += key.substr(letpos,1);
        }
        return sentence;
    }